src/read.rs

```diff
@@ -187,8 +187,8 @@ pub fn read_bytearray<R: Read>(reader: &mut R) -> Result<Vec<u8>> {
 pub fn read_position<R: Read>(reader: &mut R) -> Result<(i32, i32, i32)> {
     let val = read_u64(reader)?;
     let mut x = (val >> 38) as i32;
-    let mut y = ((val >> 26) & 0xfff) as i32;
-    let mut z = (val << 38 >> 38) as i32;
+    let mut y = (val & 0xfff) as i32;
+    let mut z = (val << 26 >> 38) as i32;
```

Comment:
Hmm, did the position data type change?

Reply:
I couldn't actually find the old encoding on wiki.vg, but prismarineJS's protocol files show this change as well:

[1.13.2](https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/1.13.2/protocol.json#L149)
[1.15.2](https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/1.15.2/protocol.json#L149)